### ISSUE TITLE
Non-stop "An unexpected error occurred" messages

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitrpg-ios#972]
User report:

> The red “An unexpected error occurred” error message has been repeatedly flashing at the bottom of the app screen and my dailies/habits aren’t resetting. I tried checking something off my to do list and it worked, and I was able to post in guild chat, but clicking the + on a habit does nothing (and they are still showing my streaks from yesterday). I have tried:
> - closing and reopening the app
> - restarting my phone
> - clearing my cache (button in menu > settings)
> - force reloading the app (button in menu > settings)
> - checking in to and out of the tavern
> - logging in to the website on my phone browser
> - logging in to the website on my computer browser
> I’m hoping this is an easy fix, I’ve been enjoying using the app and just got a couple friends to download it yesterday so they can join my party.

iOS Version: 13.2.3
Device: iPhone 8
App Version: 3.1.3 (532)
User ID: a6ba675e-ab8a-4025-8aeb-75bce0ff6669
Level: 14
Class: rogue
Is in Inn: false
Uses Costume: true
Custom Day Start: 0

**Comment by saraolson** [comment]
> The dailies aren’t resetting
> I kept pressing “Start my day” with some of the uncompleted tasks but a message saying “Unexpected error occured” kept popping out and the list with the “Start my day” popup kept appearing
> 
> I ticked off everything and it closed however the error message “unexpected error occurrd” kept appearing at the bottom of the screen every second
> 

The following lines help us find and squash the Bug you encountered. Please do not delete/change them.
iOS Version: 14.7.1
Device: iPhone 7
App Version: 3.4 (621)
User ID: a255f1ea-0e6f-49d4-8aa0-b9c69523f68e
Level: 11
Class: warrior
Is in Inn: false
Uses Costume: false
Custom Day Start: 0

**Comment by saraolson** [comment]
had a user report this and they found a way to fix it. the error carried over to when they logged into habitica on their pc but apparently refreshing there eventually cleared it up and they could use the app again normally.

> The error message kept appearing even after I force close the app (I’m on iPhone OS)
I tried closing the app and opening it again 2 times but it kept appearing.
I also uninstalled the app and reinstalled it then logged in using the account and it stayed the same with the error messages.
I can normally check the tasks off when I pressed “Start my day” and the popout went away for a second and I can also check off the tasks on the popout page normally
I eventually checked off all tasks on the popout page to stop it from appearing but instead the “An Unexpected error occured” error message kept showing up every second.
I then opened up my PC and logged in to Habitica from there and it showed the error message “An unexpected error occured” just once and I refreshed the page a couple times and it went back to normal. I checked off the tasks normally on the web page and then returned to my app to find it returned to normal. So there was no more error messages on the app after I refreshed the login on the web page.
